php/libraries/Image.class.inc

```diff
@@ -0,0 +1,256 @@
+ */
+namespace LORIS;
+
+/**
+ * This class defines a Image to allows one to type hint function's parameters
+ *
+ *  @category Imaging
+ *  @package  Main
+ *  @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+class Image
+{
+    private $_fileid;
+
+    private $_filename;
+
+    private $_filelocation;
+
+    /**
+     * Constructor
+     *
+     * @param ?int $fileid This File unique id
+     */
+    public function __construct(?int $fileid)
+    {
+        if ($fileid !== null) {
+            $dbrow = \NDB_Factory::singleton()
+                ->database()
+                ->pselectRow(
+                    '
+                     SELECT
+                       f.File as filelocation,
+                       f.OutputType as outputtype,
+                       mst.Scan_type as acquisitionprotocol,
+                       FileType as filetype
+                     FROM
+                       files f 
+                     JOIN mri_scan_type mst
+                       ON (mst.ID=f.AcquisitionProtocolID)
+                     WHERE f.FileID = :v_fileid
+                    ',
+                    array('v_fileid' => $fileid)
+                );
+            if (empty($dbrow)) {
+                throw new \NotFound('There is no file with that id');
+            }
+
+            $this->_fileid       = $fileid;
+            $this->_filename     = basename($dbrow['filelocation']);
+            $this->_filelocation = $dbrow['filelocation'];
+            $this->_outputtype   = $dbrow['outputtype'];
+            $this->_acquisitionprotocol = $dbrow['acquisitionprotocol'];
+            $this->_filetype            = $dbrow['filetype'];
+        }
+    }
+
+    /**
+     * Get file headers from paramter file
+     *
+     * @param string $headername $the parameter name to reteive
+     *
+     * @return string The value or an empty string
+     */
+    public function getHeader(string $headername): string
+    {
+        $dbrow = \NDB_Factory::singleton()
+             ->database()
+                ->pselectRow(
+                    '
+                       SELECT
+                         Value
+                       FROM
+                         parameter_file pf 
+                       JOIN parameter_type pt
+                         USING (ParameterTypeID)
+                       JOIN files f
+                         USING (FileID)
+                       WHERE
+                         f.FileID = :v_fileid AND
+                         pt.Name = :v_headername
+                      ',
+                    array(
+                     'v_fileid'     => $this->_fileid,
+                     'v_headername' => $headername,
+                    )
+                );
+        return $dbrow['Value'] ?? '';
+    }
+
+    /**
+     * Creates a imutable object representation of this tarchive
```

Comment:
Creates **an immutable** object representation of this **file** (or replace my `file` suggestion by `image` if you prefer)